discordbot/OverworldDiscordBot.py

```diff
@@ -0,0 +1,282 @@
+        _channel_id = None
+        if destination == Channels.NotificationChannelForInfo:
+            _channel_id = self.config["NotificationChannelForInfo"]
+        elif destination == Channels.EventNotificationChannelIdForShiny:
+            _channel_id = self.config["EventNotificationChannelIdForShiny"]
+        elif destination == Channels.EventNotificationChannelIdForBrilliant:
+            _channel_id = self.config["EventNotificationChannelIdForBrilliant"]
+        elif destination == Channels.EventNotificationChannelIdForMark:
+            _channel_id = self.config["EventNotificationChannelIdForMark"]
+        elif destination == Channels.EventNotificationChannelIdForRareMark:
+            _channel_id = self.config["EventNotificationChannelIdForRareMark"]
+        elif destination == Channels.TextNotificationChannelIdForShiny:
+            _channel_id = self.config["TextNotificationChannelIdForShiny"]
+        elif destination == Channels.TextNotificationChannelIdForBrilliant:
+            _channel_id = self.config["TextNotificationChannelIdForBrilliant"]
+        elif destination == Channels.TextNotificationChannelIdForMark:
+            _channel_id = self.config["TextNotificationChannelIdForMark"]
+        elif destination == Channels.TextNotificationChannelIdForRareMark:
+            _channel_id = self.config["TextNotificationChannelIdForRareMark"]
+        if _channel_id:
+            channel = self.get_channel(int(_channel_id))
+            print("{}: {}".format(channel.name, message))
+            await channel.send(message)
+        else:
+            print("LocalOnly: {}".format(message))
+
+    # function to be run in the new thread after start is called
+    def reader_func(self,ctx):
+        # config to refactor out later
+        # color for the line on the side of the embed, 0xfad1ff is pink
+        embed_color = 0xfad1ff
+        # used to verify that the read mons are new
+        last_check = 0
+        while True:
+            # check if we are supposed to be running
+            if not self.thread_running:
+                exit()
+            # refresh KCoords block
+            try:
+                self.reader.KCoordinates.refresh()
+            except Exception:
+                print("No connection to Switch at IP {}. Check that the Switch is available.".format(self.config["IP"]))
+                break
+            # read pokemon
+            pkms = self.reader.KCoordinates.ReadOwPokemonFromBlock()
+            # check if the read pokemon are different
+            if len(pkms) > 0 and pkms[-1].ec != last_check:
+                # set last check
+                last_check = pkms[-1].ec
+                # for each pkm check for filter
+                for pkm in pkms:
+                    # Determine if the pokemon info should be an event notification, and where it routes to. Send text notifications inline.
+                    _channels_to_notify = []
```

Comment:
do these variables need to be marked as internal/private when they are only accessible from inside the thread and have no use outside of it?

Reply:
I like the convention of differentiating from class variables that are members and local throwaways. Is there a common python convention for this?

Reply:
using _ itself as a variable indicates something to throwaway that isnt used at all, using it to prepend variables is typically used to mark a variable as private, or something that should not be accessed outside of the class (``self._secret_variable = 123``)

Reply:
having variables named like this is fine, but it's better to be consistent across the project and i havent used that syntax elsewhere

Reply:
OK I can remove that for consistency.